src/utils.ts

```diff
@@ -266,6 +274,176 @@ export const getConfigFilesChangedInPR = async (pr: devOps.PullRequestProperties
+           const backup = fileChanged;
+           while (fileChanged.startsWith("specification")) {
+             if (fileChanged.toLowerCase().endsWith("readme.md") && fs.existsSync(fileChanged)) {
+               if (configFiles.has(fileChanged)) {
+               } 
+               else {
+                 configFiles.set(fileChanged, []);
+               }
+               if (!backup.endsWith("readme.md")) {
+                 let value = configFiles.get(fileChanged);
+                 if (value) {
+                   value.push(backup);
+                 }
+               }
+               break;
+             }
+             // select parent readme
+             const parts = fileChanged.split("/");
+             parts.pop();
+             parts.pop();
+             parts.push("readme.md");
+             fileChanged = parts.join("/");
+           }
+         }
+         return configFiles;
+       };
+
+/**
+ * return [["Sevice/readme.md",["tag1","tag2"]]...]
+ * the tags are sorted by refered changed file's count 
+ */
+export const getTagsFromChangedFile = async (
+         filesChanged: string[]
+       ): Promise<Map<string, string[]>> => {
+  try {
+    console.log(filesChanged);
+    // traverse up to readme.md files
+    const configFiles = await getChangeFilesReadmeMap(filesChanged);
+
+    const tagsAffectedMap = new Map<string, string[]>();
+
+    configFiles.forEach((changedFiles, key) => {
+      const content = fs.readFileSync(key, { encoding: "utf8" });
+      const readme = parse(content);
+      const relativePath = getReadMeRelativePathToRepoWithoutFileName(key);
+      /**
+      *  count the changed file count for each tags
+      */
+      const tagsCnt = new Map<string, number>();
+      changedFiles.map((changedFile) => {
+        const tags = getTagsForFilesChanged(readme, [
+          changedFile.substring(changedFiles.indexOf(relativePath)),
+        ]);
+        tags.forEach((element) => {
+          const oldCnt = tagsCnt.get(element);
+          tagsCnt.set(element, oldCnt ? oldCnt + 1 : 1);
+        });
+      });
+      
+      /**
+       * first sort by count, then by tag
+       */
+      const sortedTagsCnt = [...tagsCnt].sort((a, b) => {
+        if (a[1] - b[1] === 0) {
+          if (a[0] < b[0]) {
+            return 1;
+          } else if (a[0] > b[0]) {
+            return -1;
+          }
+          return 0;
+        }
+        return b[1] - a[1];
+      });
+
+      let AffectedTags: string[] = [];
+      sortedTagsCnt.forEach((v) => {
+        if (!changedFiles.length) {
+          return;
+        }
+        const tag = v[0];
+        let tagFiles = getInputFilesForTag(readme.markDown, tag);
+
+        if (tagFiles) {
+          let intersection = tagFiles.filter((v) =>changedFiles.includes(relativePath + v));
+          if (intersection) {
+            changedFiles = changedFiles.filter((v) =>!intersection.includes(v.substring(relativePath.length)));
```

Comment:
Why reassign changedFiles

Reply:
it's for break the loop , and there is no impact to the origin map